### PR TITLE
Fixed group creation with default settings

### PR DESCRIPTION
**Pull request by harinda05**
Fix for the issue #4783

Now the enabling/disabling the OK button with regards to the text fields have been validated properly without changing the logic of the existing code. As @Siedlerchr mentioned now, no need to switch to another radio button and back again to enable the OK button.

----

- [x] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [x] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/main/java/org/jabref/gui/groups/GroupDialog.java [hunk 1]
```diff
@@ -135,6 +136,19 @@ public GroupDialog(DialogService dialogService, BasePanel basePanel, JabRefPrefe
             this.setTitle(Localization.lang("Edit group"));
         }
 
+        nameField.textProperty().addListener((observable, oldValue, newValue) -> {
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/groups/GroupDialog.java]: Can you please move these lines further down to https://github.com/JabRef/jabref/pull/4801/files#diff-2afbba630cb28bced7c30efa7c0a24dfR362, where similar listeners are registered for the other controls. Please also remove the `onAction` listener for these three controls as they are now obsolete.

File: CHANGELOG.md [hunk 2]
```diff
@@ -99,6 +99,8 @@ We refer to [GitHub issues](https://github.com/JabRef/jabref/issues) by using `#
 - We fixed an issue where only one PDF file could be imported [#4422](https://github.com/JabRef/jabref/issues/4422)
 - We fixed an issue where "Move to group" would always move the first entry in the library and not the selected [#4414](https://github.com/JabRef/jabref/issues/4414)
 - We fixed an issue where an older dialog appears when downloading full texts from the quality menu. [#4489](https://github.com/JabRef/jabref/issues/4489)
+- We fixed an issue where the enabling/disabling OK button in Create/Edit group was not properly validated with the text boxes present in the window - Group creation not possible anymore with default settings [#4783]
```
Comment (tobiasdiez) [inline CHANGELOG.md]: Please remove the changelog entry. This bug was introduced in the dev version and we use the changelog only to inform about changes relative to the latest release (I'll update the PR template to makes this more clear).

File: src/main/java/org/jabref/gui/groups/GroupDialog.java (hunk 1 above)
Comment (harinda05) [reply src/main/java/org/jabref/gui/groups/GroupDialog.java]: I'm really sorry that I didn't notice that there's someone else working on this. I have made the necessary changes you've requested. 
Regards,
Harinda

File: CHANGELOG.md (hunk 2 above)
Comment (harinda05) [reply CHANGELOG.md]: The Changelog entry is removed.

File: src/main/java/org/jabref/gui/groups/GroupDialog.java [hunk 3]
```diff
@@ -345,6 +347,32 @@ public GroupDialog(DialogService dialogService, BasePanel basePanel, JabRefPrefe
             return null;
         });
 
+        nameField.textProperty().addListener(new ChangeListener<String>() {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/groups/GroupDialog.java]: You can simplify the changeLIstener with lambdas: 
```suggestion
        nameField.textProperty().addListener((observable, oldValue, newValue) -> upddateComponent());
```

**Comment by Siedlerchr** [comment]
Please fix the remaining isuses mentioned by @tobiasdiez  so we can merge (e.g. remove the action listeners) so we can merge

File: src/main/java/org/jabref/gui/groups/GroupDialog.java (hunk 3 above)
Comment (harinda05) [reply src/main/java/org/jabref/gui/groups/GroupDialog.java]: Hi, Thanks for the suggestion. Working on it now. Will update soon

Comment (harinda05) [reply src/main/java/org/jabref/gui/groups/GroupDialog.java]: @Siedlerchr The changes have been committed :)

**Comment by LinusDietz** [comment]
Can you please give the PR a shorter and more decriptive name?

**Comment by harinda05** [comment]
> Can you please give the PR a shorter and more decriptive name?

Done :)

File: src/main/java/org/jabref/gui/groups/GroupDialog.java [hunk 4]
```diff
@@ -345,6 +345,10 @@ public GroupDialog(DialogService dialogService, BasePanel basePanel, JabRefPrefe
             return null;
         });
 
+        nameField.textProperty().addListener((observable, oldValue, newValue) -> updateComponents());
+        keywordGroupSearchTerm.textProperty().addListener((observable, oldValue, newValue) -> updateComponents());
+        searchGroupSearchExpression.textProperty().addListener((observable, oldValue, newValue) -> updateComponents());
+
         EventHandler<ActionEvent> actionHandler = (ActionEvent e) -> updateComponents();
         nameField.setOnAction(actionHandler);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/groups/GroupDialog.java]: Are these obsolete now?

**Pull request merged**